src/server/game/Handlers/ItemHandler.cpp

```diff
@@ -1250,3 +1229,153 @@ void WorldSession::HandleRemoveNewItem(WorldPackets::Item::RemoveNewItem& remove
         item->SetState(ITEM_CHANGED, _player);
     }
 }
+
+// This anonymous namespace contains utility functions for handling BagAutoSort.
+namespace
```

Comment:
this is not how c++ works, you are supposed to declare the functions in header and implement them in cpp not doing both. Using anonymouse namespaces also feels super weird, is there any benefit in it?
Also these functions are not really fitting to WorldSession due to not doing packet stuff so i'd move them to Player class instead

Reply:
It is fine (and good practice) to avoid declaring functions in headers if no other compilation units need them, which is the case here. 

As far as the anonymous namespace goes, no real reason, just convenience and making it collapsible in an IDE.